File: golang/internal/application/repository/market_price_repository.go
package repository

import (
	"folio/codinginterview/internal/domain"

	"github.com/shopspring/decimal"
)

// MarketPriceRepository は市場価格のリポジトリインターフェースです。
type MarketPriceRepository interface {
	All() (map[domain.StockSymbol]decimal.Decimal, error)
	Update(prices map[domain.StockSymbol]decimal.Decimal) error
}

File: typescript/src/application/repository/marketPriceRepository.ts
import Decimal from "decimal.js";
import { StockSymbol } from "../../domain/stockSymbol.js";

/** 市場価格リポジトリ。 */
export interface MarketPriceRepository {
  all(): Promise<Map<StockSymbol, Decimal>>;
  update(prices: Map<StockSymbol, Decimal>): Promise<void>;
}
